osquery/logger/plugins/aws_firehose.cpp

```diff
@@ -105,20 +105,17 @@ Status FirehoseLogForwarder::setUp() {
   }
 
   // Make sure we can connect to designated stream
-  Aws::Firehose::Model::ListDeliveryStreamsRequest r;
-  auto result = client_->ListDeliveryStreams(r).GetResult();
-  std::vector<std::string> stream_names = result.GetDeliveryStreamNames();
-
-  if (std::find(stream_names.begin(),
-                stream_names.end(),
-                FLAGS_aws_firehose_stream) == stream_names.end()) {
-    return Status(1,
-                  "Could not find Firehose delivery stream: " +
-                      FLAGS_aws_firehose_stream);
+  Aws::Firehose::Model::DescribeDeliveryStreamRequest r;
+  r.SetDeliveryStreamName(FLAGS_aws_firehose_stream);
+  auto result = client_->DescribeDeliveryStream(r).GetResult();
+  auto description = result.GetDeliveryStreamDescription();
+  auto stream_name = description.GetDeliveryStreamName();
+  if (stream_name.empty()){
```

Comment:
I like the idea of being explicit about asking for a specific stream and seeing if we can get it rather than asking for all streams and seeing if the one we want is in that list. However, not being familiar with this API at all, if the requested stream doesn't exist, will any of these methods returns null pointers, etc. such that a seg fault or something similar could occur with these unverified dereferences?